core/src/com/unciv/logic/automation/ConstructionAutomation.kt

```diff
@@ -94,14 +100,24 @@ class ConstructionAutomation(val cityConstructions: CityConstructions){
             NotificationIcon.Construction
         )
         cityConstructions.currentConstructionFromQueue = chosenConstruction
+
+        if (civInfo.isPlayerCivilization()) return // don't want the ai to control what a player uses faith for
+
+        val chosenItem = faithConstruction.asSequence()
+            .filterNotNull()
+            .filter { it.getStatBuyCost(cityInfo, stat = Stat.Faith)!! <= civInfo.religionManager.storedFaith }
+            .firstOrNull() ?: return
+
+
+        cityConstructions.purchaseConstruction(chosenItem.name, -1, false, stat=Stat.Faith)
```

Comment:
For now yes, BUT when the AI can choose to buy other things with faith Military units and buildings, the faith buying list will morose resemble the building construction queue

Reply:
Maybe just add say a 90% chance that the AI *won't* choose to buy *anything* (and shuffle the order when it does buy something)? That way, it will semi-organically (randomly) have a chance to save up over several turns.

IIRC I saw a random number generator with a seed stored in the save file somewhere, which should prevent save scumming from being a concern.

Reply:
Maybe add say a 90% chance that the AI won't choose to buy *anything* each turn (and randomly shuffle the order when it does buy something)? That way, it will semi-organically (randomly) have a chance to save up for more expensive purchases.

I think I saw a (presumably deterministic) random number generator somewhere that uses a seed stored in the save file, which should prevent save scumming concerns with random behaviour.